glusterd2/commands/snapshot/snapshot-create.go

```diff
@@ -610,6 +611,9 @@ func duplicateVolinfo(vol, v *volume.Volinfo) {
 	v.DistCount = vol.DistCount
 	v.Type = vol.Type
 
+	v.Metadata = make(map[string]string)
+	v.Metadata = vol.Metadata
```

Comment:
Same here.

Reply:
`vol.Metadata` is already handled during volume creation time, this wont become nil